service/history/workflow/mutable_state_impl.go

```diff
@@ -349,10 +349,7 @@ func NewMutableStateFromDB(
 	dbRecord *persistencespb.WorkflowMutableState,
 	dbRecordVersion int64,
 ) (*MutableStateImpl, error) {
-	startTime := time.Time{}
-	if dbRecord.ExecutionState.StartTime != nil {
-		startTime = dbRecord.ExecutionState.StartTime.AsTime()
-	}
+	startTime := timestamp.TimeValue(dbRecord.ExecutionState.StartTime)
```

Comment:
I guess the logic doesn't really matter here, the startTime can be empty anyway. 
If we really want to provide the right value and not relying on the overwrite logic on L416 & 418, we also need to check executionInfo.StartTime if the one in executionState is empty.

Reply:
lets not rely on executioninfo.starttime that we are planning to deprecate :)